### PR TITLE
HAAR-1630: introduce Spring type converters

### PR DESCRIPTION
**Pull request by AlTurner-MOJ**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/converter/ClientConverter.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        clientId = clientId,
+        clientIdIssuedAt = java.time.Instant.now(),
+        clientSecretExpiresAt = null,
+        clientName = clientName,
+        clientAuthenticationMethods = org.springframework.security.oauth2.core.ClientAuthenticationMethod.CLIENT_SECRET_BASIC.value,
+        authorizationGrantTypes = org.springframework.security.oauth2.core.AuthorizationGrantType.CLIENT_CREDENTIALS.value,
+        scopes = scopes,
+        clientSettings =
+        org.springframework.security.oauth2.server.authorization.settings.ClientSettings.builder()
+          .requireProofKey(false)
```
Comment (ravmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/converter/ClientConverter.kt:26]: add as static imports

File: src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/converter/ClientConverter.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  override fun convert(source: ClientCredentialsRegistrationRequest): Client? {
+    with(source) {
+      return Client(
+        id = java.util.UUID.randomUUID().toString(),
+        clientId = clientId,
+        clientIdIssuedAt = java.time.Instant.now(),
+        clientSecretExpiresAt = null,
+        clientName = clientName,
+        clientAuthenticationMethods = org.springframework.security.oauth2.core.ClientAuthenticationMethod.CLIENT_SECRET_BASIC.value,
+        authorizationGrantTypes = org.springframework.security.oauth2.core.AuthorizationGrantType.CLIENT_CREDENTIALS.value,
```
[the remark this answers is not in the document]
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/converter/ClientConverter.kt]: Done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/converter/ClientConverter.kt (hunk 1 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/converter/ClientConverter.kt:28]: Done

**Review by ravmoj — APPROVED** [review]
lgtm

**Pull request merged**